File: 02_스택, 큐, 덱/라이브 코딩/10828.py
N = int(input())
stack = []
ans = ""
for _ in range(N):
    input_list = input().split()
    command = input_list[0]

    if command == 'push':
        stack.append(int(input_list[1]))
    elif command == 'pop':
        if len(stack) == 0:
            ans += str(-1) + '\n'
        else:
            ans += str(stack.pop()) + '\n'
    elif command == 'size':
        ans += str(len(stack)) + '\n'
    elif command == 'empty':
        if len(stack) == 0:
            ans += str(1) + '\n'
        else:
            ans += str(0) + '\n'
    else:
        if len(stack) == 0:
            ans += str(-1) + '\n'
        else:
            ans += str(stack[-1]) + '\n'

print(ans)

File: 02_스택, 큐, 덱/라이브 코딩/10828.cpp
/*Ω∫≈√*/
#include <iostream>
#include <stack>

using namespace std;

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);
	
	int n, x;
	stack<int> s;
	string input;

	cin >> n;
	while (n--) {
		cin >> input;
		if (input == "push") {
			cin >> x;
			s.push(x);
		}
		else if (input == "pop") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.top() << '\n';
				s.pop();
			}
		}
		else if (input == "size") {
			cout << s.size() << '\n';
		}
		else if (input == "empty") {
			cout << s.empty() << '\n';
		}
		else if (input == "top") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.top() << '\n';
			}
		}
	}
}




